src/main/java/org/terasology/gookeeper/system/BreedingSystem.java

```diff
@@ -328,10 +346,29 @@ private Material getOffspringMaterial(EntityRef parent1, EntityRef parent2) {
             return parent1Mat;
         }
 
-        offspringGooeyColor.color.alterRed(parent1Color.color.r() + parent2Color.color.r());
-        offspringGooeyColor.color.alterBlue(parent1Color.color.b() + parent2Color.color.b());
-        offspringGooeyColor.color.alterGreen(parent1Color.color.g() + parent2Color.color.g());
+        int redValue = parent1Color.color.r() + parent2Color.color.r();
+        int blueValue = parent1Color.color.b() + parent2Color.color.b();
+        int yellowValue = parent1Color.color.g() + parent2Color.color.g();
+
+        offspringGooeyColor.color = new Color(redValue, yellowValue, blueValue);
+        String materialName = "";
+        if (redValue > 0f) {
+            materialName += ((int)(Math.ceil(offspringGooeyColor.color.r()/128))) + "r";
+        }
+        if (yellowValue > 0f) {
+            materialName += ((int)(Math.ceil(offspringGooeyColor.color.g()/128))) + "y";
+        }
+        if (blueValue > 0f) {
+            materialName += ((int)(Math.ceil(offspringGooeyColor.color.b()/128))) + "b";
+        }
+
+        offspringEntity.addOrSaveComponent(offspringGooeyColor);
+        offspringMaterial = Assets.getMaterial(materialName).get();
 
-        return offspringMaterial;
+        if (offspringMaterial != null) {
+            return offspringMaterial;
+        } else {
+            return null;
+        }
```

Comment:
This can be simplified to `Assets.getMaterial(materialName).orElse(null)`

Additionally as per [the `get` javadocs](https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#get) the call will throw an error if the value was null. Judging by how you are handling the case where the value is null this is probably something to consider